sql/src/main/java/org/apache/druid/sql/calcite/parser/DruidSqlParserUtils.java

```diff
@@ -70,13 +75,76 @@
 import java.util.ArrayList;
 import java.util.Arrays;
 import java.util.List;
+import java.util.Map;
 import java.util.stream.Collectors;
 
 public class DruidSqlParserUtils
 {
   private static final Logger log = new Logger(DruidSqlParserUtils.class);
   public static final String ALL = "all";
 
+  @VisibleForTesting
+  public static final String PARTITION_ERROR_MESSAGE =
+      "Invalid granularity [%s] after PARTITIONED BY.  "
+      + "Expected HOUR, DAY, MONTH, YEAR, ALL TIME, FLOOR() or TIME_FLOOR()";
+
+  public enum GranularityGrain
+  {
+    SECOND_GRAIN("SECOND", Granularities.SECOND),
+    MINUTE_GRAIN("MINUTE", Granularities.MINUTE),
+    HOUR_GRAIN("HOUR", Granularities.HOUR),
+    DAY_GRAIN("DAY", Granularities.DAY),
+    WEEK_GRAIN("WEEK", Granularities.WEEK),
+    MONTH_GRAIN("MONTH", Granularities.MONTH),
+    QUARTER_GRAIN("QUARTER", Granularities.QUARTER),
+    YEAR_GRAIN("YEAR", Granularities.YEAR),
+    ALL_GRAIN("ALL", Granularities.ALL),
+    ALL_TIME_GRAIN("ALL TIME", Granularities.ALL);
+
+    static final Map<String, GranularityGrain> VALUE_TO_GRANULARITY_GRAIN = ImmutableMap.of(
```

Comment:
Can we not use `Granularity.fromString()` with the supplied literal and not have this reverse-mapping/enum at all?

Reply:
it seems to me that there are a few aliases here - but it would be great to rely on the existing classes - there is a `GranularityType` enum which seems to be a close relative to what `GranularityGrain` describes

Reply:
Thanks, using GranularityType now